docs/design/collaboration/Building the collaboration paradigm between App Developers and Platform Developers with Kusion.md

```diff
@@ -0,0 +1,235 @@
+
+### Cons
+
+- Most of the work was handed over to platform developers. Different applications may have different requirements for the same resource attribute, and platform developers' workload has increased several times.
+It is reasonable to allow app developers to configure certain attributes, such as the database engine version since it can impact their business logic code.
+
+## Summary
+
+We won't list all the projects we have researched here. Each project has its advantages and we have learned a lot from them, but none of them can meet the goals defined above. The following section will introduce our proposal
+
+# Our Design
+
+According to our experiences and extensive conversations with other companies, teams included in SLCD(Software Life Cycle Development) are generally divided into two parts: platform engineers and application engineers. 
+
+- Application engineers mainly focus on business logic and choose infrastructures to support them in achieving their business goals.
+- Platform engineers are responsible for infrastructure standardization which helps to reduce the cognitive load of application engineers and meet their requirements. This, in turn, boosts the productivity of the organization.
+
+We aim to minimize application developers' cognitive load and enable self-service so we do not let application developers care about low-level infrastructure details. They only need to declare what feature they want in a developer-friendly way, such as declaring one application needs a MySQL database and public Internet access with Kusion models provided by platform developers. Then Kusion will provision a MySQL database and a LoadBalancer that complies with platform standardization.
+
+We introduce a set of Kusion models to standardize deployment and the root model is AppConfiguration. AppConfiguration is a declarative application configuration model maintained by application developers to configure all operation intentions during SLCD and more details can be found on our [website](https://kusionstack.io/docs/user_docs/concepts/appconfiguration). Each Kusion model in the AppConfiguration represents a building block built by platform developers to standardize the application's behavior and hide infrastructure complexity, such as database, networking, and monitoring.
+
+## Terms
+
+Before we dive into details, it is necessary to clarify certain terms.
+- Environment: An environment is a logical concept representing a space where applications can be deployed. It could be a stage during your deployment such as `dev`,`staging`, and `prod` or cloud vendors such as `AWS`,`Azure`, and `Aliyun` or a combination of the two.
+- [Project](https://kusionstack.io/docs/user_docs/concepts/glossary#project): A project is a logical concept to bundle application configurations.
+- [Stack](https://kusionstack.io/docs/user_docs/concepts/glossary#stack): A stack is a logical concept represents where you execute a Kusion command.
```

Comment:
I think stack needs a better definition than " a place to execute kusion command"